### PR TITLE
Use automatic task dependencies

### PR DESCRIPTION
**Pull request by wolfs**
It seems like the manual `dependsOn` calls are mostly unnecessary.

### CODE REVIEW

File: buildSrc/src/main/java/JniPlugin.java [hunk 1]
```diff
@@ -1,92 +1,38 @@
...
-        RemoveGeneratedNativeHeaders removeGeneratedNativeHeaders,
         TaskProvider<JavaCompile> compileJavaProvider
     ) {
         compileJavaProvider.configure(compileJava -> {
-            compileJava.getOptions().getCompilerArgumentProviders().add(compilerArguments);
-            // Cannot do incremental header generation
+            DirectoryProperty headerOutputDirectory = compileJava.getOptions().getHeaderOutputDirectory();
+            headerOutputDirectory.convention(compileJava.getProject().getLayout().getBuildDirectory().dir("generated/jni-headers"));
+            // The nested output is not marked automatically as an output of the task regarding task dependencies.
+            // So we mark it manually here.
```
Comment (lptr) [inline buildSrc/src/main/java/JniPlugin.java]: Can we link to an issue here?

File: buildSrc/src/main/java/JniPlugin.java [hunk 2]
```diff
@@ -1,92 +1,38 @@
...
+            compileJava.getOutputs().dir(compileJava.getOptions().getHeaderOutputDirectory());
+            // Cannot do incremental header generation, since the pattern is currently wrong.
```
Comment (lptr) [inline buildSrc/src/main/java/JniPlugin.java]: Link to issue here too?

File: buildSrc/src/main/java/JniPlugin.java (hunk 1 above)
Comment (wolfs) [reply buildSrc/src/main/java/JniPlugin.java]: Added.

File: buildSrc/src/main/java/JniPlugin.java (hunk 2 above)
Comment (wolfs) [reply buildSrc/src/main/java/JniPlugin.java]: Added.

**Comment by wolfs** [comment]
@lptr PTAL!

**Comment by lptr** [comment]
LGTM.

**Pull request merged**